Token_help.md

```diff
@@ -0,0 +1,57 @@
+## Token Generation
+- In cmake test suit we have integrated a new feature to generate token and parent token and the compare the output and result file. If both the files are same then test case will pass otherwise testcase will be failed.
+- Generating Token files 
+- we can generate Token files by using the  command line option ```--token``` followed by token file name
+	- Eg: ``` uncrustify.exe -c Uncrustify.Common-CStyle.cfg -p debug.txt -f input.cpp -o output.cpp --token Tokens.txt -La  > a.log 2>&1```
+	
+	 ```
+		 Line [Token]             Text
+		 1    [PREPROC]           #
+		 1    [PP_INCLUDE]        include
+		 1    [PREPROC_BODY]      "macros.h"
+		 1    [NEWLINE]           
+		 3    [FUNC_CALL]         MACRO_DEFINED_ELSEWHERE
+		 3    [FPAREN_OPEN]       (
+		 3    [WORD]              Argument
+		 3    [FPAREN_CLOSE]      )
+		 3    [BRACE_OPEN]        {
+		 3    [NEWLINE]           
+		 4    [BRACE_CLOSE]       }```
+		 
+
+- we can generate Parent_Token files by using the  command line option ```--parent``` followed by token file name
+	- Eg: ``` uncrustify.exe -c Uncrustify.Common-CStyle.cfg -p debug.txt -f input.cpp -o output.txt --parent parent_token.txt -La  > a.log 2>&1```
+		
+		```
+			Line [Parent]            Text
+			1    [PP_INCLUDE]        #
+			1    [NONE]              include
+			1    [NONE]              "macros.h"
+			1    [NONE]              
+			3    [NONE]              MACRO_DEFINED_ELSEWHERE
+			3    [FUNC_CALL]         (
+			3    [NONE]              Argument
+			3    [FUNC_CALL]         )
+			3    [FUNC_CALL]         {
+			3    [NONE]              
+			4    [FUNC_CALL]         }
+		```
+
+- we can generate generate Tokens & Parent Tokens on one go by enabling both --token,--parent
+	- Eg: ``` uncrustify.exe -c Uncrustify.Common-CStyle.cfg -p debug.txt -f ip.cpp -o op.txt --token Tokens.txt --parent parent_token.txt -La  > a.log 2>&1```
+
+### Cmake Changes for RUN_TESTS
```

Comment:
I have no idea what this section is about and why it's needed

Reply:
This is just an example for generating tokens and parent tokens both.

Reply:
No, I meant below